scalafix-interfaces/src/main/java/scalafix/interfaces/ScalafixPatch.java

```diff
@@ -0,0 +1,9 @@
+package scalafix.interfaces;
+
+public interface ScalafixPatch {
+    /**
+     * Can be RemoveGlobalImport, RemoveImportee, AddGlobalImport, AddGlobalSymbol, ReplaceSymbol
```

Comment:
it looks like an enum

Reply:
you are right, but it's not corresponding to all possible cases in Patch sealed trait.
Creating an Enum may add too much constraint for this field, especially that the only use case that I see right now is applying only patches relate to `Import`

My opinion is that String is enough for now, but maybe later we can add an enum.

Reply:
> the only use case that I see right now is applying only patches relate to Import

Can you elaborate the use-case? maybe we don't need to expose the path class hierarchy as-is, but just what we need - I am concerned we are leaking internals here.

Reply:
If we apply a rule that looks like explicitResultType: When a type is added, sometimes an import is also added. 
We would like to be able to apply all import patches first. 
I understand your concern. We can maybe expose 3 kinds: ReplaceSymbol, AddImport, RemoveImport? 
I can also remove `kind` until it's clearer what we would need, and add this in a second PR later.

Reply:
>  We would like to be able to apply all import patches first.

I am not very familiar with the patch internals, but doesn't https://scalacenter.github.io/scalafix/docs/developers/patch.html#atomic allow to group patches that should be applied "together" - maybe we need to expose patches as group of atomic patches?

Reply:
>  I can also remove kind until it's clearer what we would need, and add this in a second PR later.

That would probably be better as there is already a lot of surface in the new APIs. `ScalafixPatch` would then become an empty trait for now, highlighting the fact that it's used for its identify (to iterate and converge on a proper state), rather than for its attributes.